### ISSUE TITLE
The table column `content`/`patch` in `issue`/`comment` should be `longtext`

### ISSUE THREAD

**Issue by wxiaoguang** [go-gitea/gitea#16656]
- Gitea version (or commit ref): 1.15.0-rc3
- Git version: N/A
- Operating system: N/A
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite

## Description

When migrating issues with long content, gitea may report error: 

```
2021/08/09 17:53:26 modules/task/task.go:54:handle() [E] Run task failed: Error 1406: Data too long for column 'content' at row 1
```

## Bypass Fix

```
alter table issue change content content longtext;
alter table `comment` change content content longtext;
alter table `comment` change patch patch longtext;
```

**Comment by lunny** [comment]
`text` column on `mysql` supports maxium 64K Bytes it seems it enough.

**Comment by wxiaoguang** [comment]
Actually there are many issues in GitHub/GitLab have very long content, eg: issues with detail logs. And we can easily find large content issue in gitea's issue list, eg:  `https://github.com/go-gitea/gitea/issues/16581#issuecomment-890123001` the size is more than 10KB.

IMO there is no necessary to limit content size. ps: Bill Gates said that "640KB is enough for everyone" 😂

If gitea must have the 64KB limit, it's better to tell users who are migrating repositories the detailed failure reason and which issue causes the failure. Otherwise end users have no knowledge how to deal this the unclear error message and can never complete the migration.

**Comment by wxiaoguang** [comment]
I proposed a PR: https://github.com/go-gitea/gitea/pull/16765

When using `TEXT` type, only MySQL has 64KB limitation, pgSQL/MSSQL/SQLite don't have such limitation.

I think is good to eliminate the limitation across databases.

**Comment by noerw** [comment]
implemented in #16765
